bindings.c

```diff
@@ -4140,10 +4213,14 @@ static struct cg_proc_stat *find_proc_stat_node(const char *cg)
 
 	do {
 		if (strcmp(cg, node->cg) == 0)
-			return node;
+			goto out;
 	} while ((node = node->next));
 
-	return NULL;
+	node = NULL;
+
+out:
+	prune_proc_stat_history();
```

Comment:
I could imagine this going wrong based on timing.  You take the lock;  find the entry, make sure it's not stale;  then you drop the lock, call prune, which takes the lock and might drop this node if it by now has become stale;  finally you return this node unconditionally, but it may have been pruned, right?

Reply:
Right, but I don't think that it can happen because of the way the function is used. Nodes are pruned when the underlying cgroup no longer exists, but since the call is made by a process from the cgroup, it has to exist and thus the node cannot be pruned.